enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+---
+title: Evidence Forwarding
+authors:
+  - rhmdnd
+  - JAORMX
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2022-07-12
+last-updated: 2022-07-12
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+---
+
+To get started with this template:
+1. **Pick a domain.** Find the appropriate domain to discuss your enhancement.
+1. **Make a copy of this template.** Copy this template into the directory for
+   the domain.
+1. **Fill out the "overview" sections.** This includes the Summary and
+   Motivation sections. These should be easy and explain why the community
+   should desire this enhancement.
+1. **Create a PR.** Assign it to folks with expertise in that domain to help
+   sponsor the process.
+1. **Merge at each milestone.** Merge when the design is able to transition to a
+   new status (provisional, implementable, implemented, etc.). View anything
+   marked as `provisional` as an idea worth exploring in the future, but not
+   accepted as ready to execute. Anything marked as `implementable` should
+   clearly communicate how an enhancement is coded up and delivered. If an
+   enhancement describes a new deployment topology or platform, include a
+   logical description for the deployment, and how it handles the unique aspects
+   of the platform. Aim for single topic PRs to keep discussions focused. If you
+   disagree with what is already in a document, open a new PR with suggested
+   changes.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+See ../README.md for background behind these instructions.
+
+Start by filling out the header with the metadata for this enhancement.
+
+# Evidence Forwarding
```

Comment:
I complained about the use of "Evidence" only since we don't use that term to refer to the ARF within the operator project (even though evidence makes sense, we always call it Raw/detailed results). Since this enhancement is separate from the CRD results, I want to be clear of that confusion with the naming that we decide to use.

Reply:
Yeah - I think that's a good point. I'm not attached to the term "evidence". I was using it in passing to refer to raw results kept in large files and it made it's way into the enhancement.

I can see how that can be confusing.

What would we call this enhancement instead? Raw Result Forwarding?

Reply:
Yes please, let's not use the word evidence. In the compliance word, evidence is used for something else we are also occasionally asked to implement, which is e.g. "this rule that was supposed to ensure that the audit logs directory is secure failed, because the audit rule directory has perms 777 and here is the output of ls -l".

Reply:
I used the word evidence initially because I didn't want to limit the implementation to only forwarding the ARF results and use the same mechanism to forward other things. e.g. the data gathered by the api-resource-collector is not really results and can be considered evidence instead.

In the future, other artifacts considered evidence could be forwarded as well via similar mechanisms. e.g. VM snapshots.